documentation/modules/exploit/linux/http/bolt_authenticated_rce.md

```diff
@@ -0,0 +1,83 @@
+## Vulnerable Application
+This module exploits multiple vulnerabilities in Bolt CMS version 3.7.0 and 3.6.* in order to execute arbitrary commands as root.
+
+This module first authenticates to Bolt CMS and visits the profile page to obtain a special token. This token is then used to change the username in /bolt/profile to a PHP `system($_GET[""])` variable, using a generated random name. If this succeeds, the target may be vulnerable, and the module proceeds by issuing an HTTP GET request for /bolt/overview/showcases in order to obtain a CSRF token to be used later.
+
+Next, the module obtains a list of filename tokens from `/async/browse/cache/.sessions.` These tokens are used to create files with the blacklisted `.php` extention via HTTP POST requests to `/async/folder/rename`. With the CSRF token obtained before, it is possible to create .php files by "renaming" these cache tokens. While most (if not all) available tokens can be used to created .php files in the /root directory on the server, the resulting files cannot always be used to execute commands. The module excludes tokens with a filesize of under 2000 bytes, as they can't ever be used for this purpose. For the remaining tokens, it seems the only way to identify a valid one, is to use each token to create a .php file and then check the contents of that file. The module does exactly this, deleting any .php files that can't be used, until it finds a file for which the "displayname" value is an empty string (""). This value indicates that the profile username matching this token is the `$_GET` variable generated earlier, and that the file can be used to execute arbitrary commands. If a valid .php file is created, the module executes the payload as root via an HTTP get request in this format: `/files/<rogue_PHP_file>?<$_GET_variable>=<payload>`
+
+The module requires valid credentials for a user with administrative privileges. This module has been successfully tested on Bolt CMS 3.7.0.
+
+## Verification Steps
+1. Install the module as usual
+2. Start msfconsole
+3. Do: `use exploit/linux/http/bolt_authenticated_rce`
+4. Do: `set RHOSTS [IP]`
+5. Do: `set USERNAME [username for the Bolt CMS account]`
+6. Do: `set PASSWORD [password for the Bolt CMS account]`
+7. Do: `set RHOSTS [IP]`
+8. Do: `set payload [payload]`
+9. Do: `set LHOST [IP]`
+10. Do: `exploit`
+
+## Options
+1. `FILE_TRAVERSAL_PATH`. This is the traversal path to get from the `/files/` directory on the web server to the `/root` directory on the server. It is used by the module to write rogue .php files to /root. The default value is `../../../public/files`.
+2. `PASSWORD`. The password for the Bolt CMS account to authenticate with. This option is 
+required.
+3. `TARGETURI`. The base path to Bolt CMS. The default value is `/`.
+4. `USERNAME`. The username for the Bolt CMS account to authenticate with. This option is required.
+
+
+## Scenario
+```
```

Comment:
Add an h3 (###) of the cms version and OS.... prob:


### Bolt CMS 3.7.0 running on CentOS 7